obs-studio-server/source/nodeobs_api.cpp

```diff
@@ -267,6 +267,65 @@ static void DeleteOldestFile(const char* location, unsigned maxLogs)
 #include <unistd.h>
 #endif
 
+outdated_driver_error * outdated_driver_error::inst = nullptr;
+
+outdated_driver_error * outdated_driver_error::instance()
+{
+	if(inst == nullptr)
+	{
+		inst = new outdated_driver_error();
+	}
+	return inst;
+}
+
+void outdated_driver_error::set_active( bool state) 
+{
+	if (state) {
+		if (!lookup_enabled)
+		{
+			line_1 = "";
+			line_2 = "";
+		}
+		lookup_enabled++;
+	} else {
+		lookup_enabled--;
+	}
+}
+
+std::string outdated_driver_error::get_error(const char* base_error)
+{
+	if (std::string(base_error).find("Function not implemented") == std::string::npos)
+		return "";
+
+	if(line_1.size() && line_2.size())
+		return line_1 + std::string("\n") + line_2;
+	else 
+		return "";
+}
+
+void outdated_driver_error::catch_error(const char* msg)
+{
+	if (!lookup_enabled)
+		return;
+
+	const std::string msg_string = msg;
+
+	if (line_1.size()==0) {
+		const std::string line_1_pattern = "Driver does not support the required nvenc API version";
```

Comment:
I would define this string in a #define, same for `const std::string line_2_pattern`

Reply:
I have thought of making this two variables explicitly static. To tell a compiler not to recreate it on every function call in case it does not do so based only on const.  
Converting to a #define will make them more loosely typed and possibly add a type conversion.